File: include/PolyWorksSDK/samples/applications/IMEdit/CSharp/CSharpWallCreation/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CSharpWallCreation
{
    public partial class MainForm : Form
    {
        // =========================================================================================
        // ========================================= Methods =======================================

        // =========================================================================================
        // Constructor
        // -----------------------------------------------------------------------------------------
        public MainForm()
        {
            InitializeComponent();
        }

        // =========================================================================================
        // Main action of this application
        // -----------------------------------------------------------------------------------------
        private void ButtonStart_Click(object sender, EventArgs e)
        {
            IMEditLib.IMEdit imEditObject = new IMEditLib.IMEdit();

            TextLog.Text = "";

            // If the server does not exists, return
            if ( imEditObject == null )
            {
                return;
            }

            // Obtain PolyWorks|Modeler's current project
            IMEditLib.IIMEditProject projectObject = null;
            imEditObject.ProjectGetCurrent( out projectObject );

            if ( projectObject == null )
            {
                imEditObject = null;
                return;
            }

            // Obtain a Command Center from the current PolyWorks|Modeler project
            IMEditLib.IIMCommandCenter imCommandCenter = null;
            projectObject.CommandCenterCreate(out imCommandCenter);

            // Send commands
            String buffer;
            int result;

            if ( imCommandCenter != null )
            {
                buffer = "FILE NEW_PROJECT";
                TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                imCommandCenter.CommandExecute( buffer );

                buffer = "TREEVIEW MODEL VIEW DEFAULT_STATIC FLAT";
                TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                imCommandCenter.CommandExecute( buffer );

                buffer = String.Concat( "FILE IMPORT_MODEL POLYGONAL_FILE ( \"", TextSource.Text, "\" )" );
                TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                result = imCommandCenter.CommandExecute( buffer );

                // If the previous operation was not a success, we're finished
                if ( imCommandCenter.ReturnValueIsSuccess( result ) != 0 )
                {
                    buffer = "EDIT PLANE CREATE XY_PLANE";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "EDIT OBJECT INVERT_ORIENTATION";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "SELECT TRIANGLES ALL";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "EDIT TRIANGLES OFFSET NORMAL ( -1.0, \"On\", \"Off\", 180.0 )";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "SELECT TRIANGLES ALL";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "EDIT TRIANGLES SLICE_WITH_A_PLANE INSERT_INTERSECTION";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "SELECT TRIANGLES USING_PLANES BELOW";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "EDIT DELETE LAST";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "TREEVIEW OBJECT VIEW HIDE";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "VIEW POSE CENTER";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );

                    buffer = "EDIT TRIANGLES CREATE_WALLS ( 3.0 )";
                    TextLog.Text = String.Concat( TextLog.Text, buffer, "\r\n" );
                    imCommandCenter.CommandExecute( buffer );
                }
            }

            // Cleanup
            System.Runtime.InteropServices.Marshal.ReleaseComObject( imCommandCenter ); 
            imCommandCenter = null;
            System.Runtime.InteropServices.Marshal.ReleaseComObject( projectObject );
            projectObject = null;
            System.Runtime.InteropServices.Marshal.ReleaseComObject( imEditObject );
            imEditObject = null;
        }
    }
}

File: include/PolyWorksSDK/samples/applications/IMEdit/VisualBasic.Net/VBNetWallCreation/MainForm.vb
Public Class MainForm

    ' Global COM Message Filter to prevent COM calls from being rejected if the server application is temporarily busy
    Private comFilter As CComMessageFilter = Nothing

    ' Upon loading the Form
    Private Sub MainForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        comFilter = New CComMessageFilter

    End Sub

    ' Cleanup upon destruction of the form
    Private Sub MainForm_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed

        comFilter = Nothing

    End Sub

    ' Main action of this application
    Private Sub ButtonStart_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ButtonStart.Click

        Dim imEditObject As New IMEditLib.IMEdit

        TextLog.Text = ""

        ' If the server does not exists, return
        If imEditObject Is Nothing Then

            Exit Sub

        End If

        ' Obtain PolyWorks|Modeler's current project
        Dim projectObject As IMEditLib.IIMEditProject = Nothing
        imEditObject.ProjectGetCurrent(projectObject)

        If projectObject Is Nothing Then

            System.Runtime.InteropServices.Marshal.ReleaseComObject(imEditObject)
            imEditObject = Nothing
            Exit Sub

        End If

        ' Obtain a Command Center from the current PolyWorks|Modeler project
        Dim commandCenterObject As IMEditLib.IIMCommandCenter
        projectObject.CommandCenterCreate(commandCenterObject)

        ' Send commands
        Dim buffer As String
        Dim result As Short

        If Not commandCenterObject Is Nothing Then

            buffer = "FILE NEW_PROJECT"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

            buffer = "TREEVIEW MODEL VIEW DEFAULT_STATIC FLAT"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            commandCenterObject.CommandExecute(buffer)

            buffer = "FILE IMPORT_MODEL POLYGONAL_FILE ( """ & TextSource.Text & """ )"
            TextLog.Text = TextLog.Text & buffer & vbCrLf
            result = commandCenterObject.CommandExecute(buffer)

            ' If the previous operation was not a success, we're finished
            If commandCenterObject.ReturnValueIsSuccess(result) <> 0 Then

                buffer = "EDIT PLANE CREATE XY_PLANE"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "EDIT OBJECT INVERT_ORIENTATION"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "SELECT TRIANGLES ALL"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "EDIT TRIANGLES OFFSET NORMAL ( -1.0, ""On"", ""Off"", 180.0 )"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "SELECT TRIANGLES ALL"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "EDIT TRIANGLES SLICE_WITH_A_PLANE INSERT_INTERSECTION"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "SELECT TRIANGLES USING_PLANES BELOW"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "EDIT DELETE LAST"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "TREEVIEW OBJECT VIEW HIDE"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "VIEW POSE CENTER"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

                buffer = "EDIT TRIANGLES CREATE_WALLS ( 3.0 )"
                TextLog.Text = TextLog.Text & buffer & vbCrLf
                commandCenterObject.CommandExecute(buffer)

            End If

        End If

        ' Cleanup
        System.Runtime.InteropServices.Marshal.ReleaseComObject(commandCenterObject)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(projectObject)
        System.Runtime.InteropServices.Marshal.ReleaseComObject(imEditObject)

        commandCenterObject = Nothing
        projectObject = Nothing
        imEditObject = Nothing

    End Sub

End Class
